controllers/capability/reconciler.go

```diff
@@ -159,6 +168,23 @@ func (r *Reconciler) updateStatefulSetIfOutdated(desiredSts *appsv1.StatefulSet)
 	return true, err
 }
 
+func (r *Reconciler) deleteStatefulSetIfOldLabelsAreUsed(desiredSts *appsv1.StatefulSet) (bool, error) {
+	currentSts, err := r.getStatefulSet(desiredSts)
+	if err != nil {
+		return false, err
+	}
+
+	if _, ok := currentSts.Labels[OldKeyActiveGate]; !ok {
```

Comment:
Is is sufficient to compare the desired labels with the ones given in the StatefulSet.